### ISSUE TITLE
cmd/compile: nil type word in interface

### ISSUE THREAD

**Issue by josharian** [golang/go#28390]
Originally reported by @shuLhan on golang-dev.

The following code panics due to a nil type word:

```go
package main

import (
	"fmt"
)

type A struct {
	K int
	S string
	M map[string]string
}

func newA(k int, s string) (a A) {
	a.K = k
	a.S = s
	a.M = make(map[string]string)
	a.M[s] = s
	return
}

func proxy() (x int, a A) {
	return 1, newA(2, "3")
}

func consume(x int, a interface{}) {
	fmt.Println(x)
	fmt.Println(a)
}

func main() {
	consume(proxy())
}
```

Introduced by CL 114797.

**Comment by josharian** [comment]
I'm really surprised that this didn't fail anywhere else in all.bash.

In any case, testing a fix now.

**Comment by cherrymui** [comment]
> I'm really surprised that this didn't fail anywhere else in all.bash.

I guess it only triggers in the form of `f(g())` where `g` returns multiple results and some of the results need conversion? That is probably quite rare...

**Comment by josharian** [comment]
Actually, `g()` is inlined in the code above, so it isn't (really) of the form `f(g())`.

**Comment by tandr** [comment]
if this regression is reproducible on Go 1.11.1 (as reported in golang-nuts), it seems to be appropriate to have cherrypick it for go1.11.2 as a target. (IMHO)

**Comment by josharian** [comment]
Do you have a link handy for the 1.11.1 reproduction report? I can’t reproduce locally with 1.11.1. And the code in question is very much new to 1.12...

**Comment by cherrymui** [comment]
I don't think it is reproducible with Go 1.11. I can only reproduce it with tip, and the culprit CL is pretty recent, past Go 1.11.

@tandr do you have a link to golang-nuts report?

**Comment by tandr** [comment]
D'oh, my bad... I misread the https://groups.google.com/forum/#!topic/golang-dev/3Fy0M5dlc-w -- it is NOT reproducible on 1.11.1

Terrible sorry for the noise
